### PR TITLE
FollowTargetGoal -> ActiveTargetGoal

### LINKED ISSUE #2560: FollowTargetGoal is not accurate
**Class Description**
`FollowTargetGoal` is the standard tracking goal used for entity AI. It first locates a nearby entity matching a predicate (if any); if one is found, the goal assigns it as the target of the `FollowTargetGoal`'s owner.

**Issue**
My issue with this name is that the goal does not actually make the entity "follow" anything, and "follow" implies action. Because the goal is based around `findClosestTarget` and setting a `target` on the mob, I think `TrackClosestTargetGoal` might be more appropriate (or something to that degree). The parent class is already named `TrackTargetGoal`, so adding a "closest" (which is pretty much the only thing `FollowTargetGoal` does) makes sense.

**Other Thoughts**
I'm guessing the "Follow" part of the name was taken from the `getFollowRange` (`GENERIC_FOLLOW_RANGE` attribute) method called inside the goal, but that seems to do more with the size of the goal rather than what it actually does.

### PR DESCRIPTION
**Pull request by frqnny**
Closes #2560, see what draylar put there.

adding on, "nearest" wouldn't fit because it does call a bunch of methods that also use "closest"

### CODE REVIEW

**Comment by liach** [comment]
Imo nearest isn't the principal feature of this goal. The distinguishing feature is this is that the goal finds targets by an entity class.

**Review by liach — CHANGES_REQUESTED** [review]
You didn't check the subclasses. Definitely need to rename `DisableableFollowTargetGoal` and `FollowTargetIfTamedGoal` if this is rename is done.

Now I think `ActiveTargetGoal` (find target actively) may be better, as this is what hostile mobs use to target players

**Comment by frqnny** [comment]
I went ahead and renamed some sub classes. For entity-specific ones like Spider, I just wrote it "TargetGoal" since it would be "SpiderEntity.TargetGoal"

File: mappings/net/minecraft/entity/ai/goal/EnableableActiveTargetGoal.mapping [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_3760 net/minecraft/entity/ai/goal/DisableableFollowTargetGoal
+CLASS net/minecraft/class_3760 net/minecraft/entity/ai/goal/EnableableActiveTargetGoal
```
Comment (liach) [inline mappings/net/minecraft/entity/ai/goal/EnableableActiveTargetGoal.mapping]: How about `Toggleable` if "enableable" and "disableable" appear too cursed

File: mappings/net/minecraft/entity/ai/goal/TameableTargetGoal.mapping [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_1404 net/minecraft/entity/ai/goal/FollowTargetIfTamedGoal
+CLASS net/minecraft/class_1404 net/minecraft/entity/ai/goal/TameableTargetGoal
```
Comment (liach) [inline mappings/net/minecraft/entity/ai/goal/TameableTargetGoal.mapping]: image
Should be `UntamedActiveTargetGoal`

File: mappings/net/minecraft/entity/passive/PolarBearEntity.mapping [hunk 3]
```diff
@@ -19,7 +19,7 @@ CLASS net/minecraft/class_1456 net/minecraft/entity/passive/PolarBearEntity
 	METHOD method_6602 playWarningSound ()V
 	METHOD method_6603 setWarning (Z)V
 		ARG 1 warning
-	CLASS class_1457 FollowPlayersGoal
+	CLASS class_1457 PlayerTargetGoal
```
Comment (liach) [inline mappings/net/minecraft/entity/passive/PolarBearEntity.mapping]: This one is like targeting to protect baby (when there is baby close around) while the other one is general "anger at" target goal

**Review by liach — APPROVED** [review]
Thank you for these names :heart:

**Comment by liach** [comment]
@modmuss50 How feasible is this bug-fixing one? Though this may affect mods using the old ai system (assuming they aren't using brains), these renames do accuracy describe the actual roles of these goals.

**Comment by modmuss50** [comment]
I dont think these classes are used that widely, so changing them in 1.17 seems fine to me. Does anyone have any objections to this?

**Review by modmuss50 — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Pull request merged**